sssd_test_framework/roles/keycloak.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+import json
+import shlex
+
+from pytest_mh.ssh import SSHProcessResult
+
+from ..hosts.keycloak import KeycloakHost
+from .base import BaseLinuxRole, BaseObject
+
+__all__ = [
+    "Keycloak",
+    "KeycloakUser",
+    "KeycloakGroup",
+]
+
+
+class Keycloak(BaseLinuxRole[KeycloakHost]):
+    """
+    Keycloak service management.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def setup(self) -> None:
+        """
+        Make sure we have admin's TGT so we can run ipa cli commands.
+        """
+        super().setup()
+        self.host.kclogin()
+
+    def kcadm(self, command: str) -> SSHProcessResult:
```

Comment:
1. Can we use svc / systemctl to manage this?
2. To conform to other classes in the framework, rename this to start? 
3. Do we need stop/restart ?

Reply:
1.  No, kcadm is the command line interface to the Keycloak API.   It's needed for all the user/group commands as well as any other commands directly interfacing with Keycloak.   The service is handled by systemctl on the server in our setup on sssd-ci-containers and other CIs.

2.  As mentioned this does more than just start the process.   It's used by all the supporting methods that interact with Keycloak.   It's more like the IPA _exec I think.   Maybe I should rename it _kcadm?   I had left it without the underscore in case we wanted to allow other keycloak commands to be executed in tests by the role.

3.  Generally, I don't think so but, would that go in the role or host class?   Do other providers have a start/stop/restart build into the role/host?

Reply:
Okay, so it's like "ipa" command, this is good then, I think. I'd update the docstring to describe that the kcadm command is. The command to interface with the keycloak API: create users, etc.